src/configure.rs

```diff
@@ -217,41 +216,26 @@ fn handle_toggle_configuration(doc: &mut Document, name: &str, key: &str) -> Res
     Ok(())
 }
 
-pub fn get_configuration() -> toml::Table {
-    let starship_config = StarshipConfig::initialize();
+pub fn get_configuration(context: &Context) -> toml::Table {
+    let starship_config = StarshipConfig::initialize(&context.get_config_path_os());
 
-    starship_config
-        .config
-        .expect("Failed to load starship config")
+    starship_config.config.unwrap_or(toml::Table::new())
 }
 
-pub fn get_configuration_edit() -> Document {
-    let file_path = get_config_path();
-    let toml_content = match utils::read_file(file_path) {
-        Ok(content) => {
-            log::trace!("Config file content: \"\n{}\"", &content);
-            Some(content)
-        }
-        Err(e) => {
-            let level = if e.kind() == ErrorKind::NotFound {
-                log::Level::Debug
-            } else {
-                log::Level::Error
-            };
-
-            log::log!(level, "Unable to read config file content: {}", &e);
-            None
-        }
-    };
+pub fn get_configuration_edit(context: &Context) -> Document {
+    let config_file_path = context.get_config_path_os();
+    let toml_content = StarshipConfig::read_config_content_as_str(&config_file_path);
 
     toml_content
         .unwrap_or_default()
         .parse::<Document>()
         .expect("Failed to load starship config")
 }
 
-pub fn write_configuration(doc: &Document) {
-    let config_path = get_config_path();
+pub fn write_configuration(context: &Context, doc: &Document) {
+    let config_path = context
+        .get_config_path_os()
+        .expect("able to get config path");
```

Comment:
This isn't really worse than what we had, but for user-facing errors like this, please prefer using `eprintln!` and `exit`, the output of `expect` can be hard to parse.

Reply:
sure, can do